operator/src/main/java/oracle/kubernetes/operator/Main.java

```diff
@@ -858,24 +866,70 @@ public NextAction onFailure(Packet packet, CallResponse<V1NamespaceList> callRes
     @Override
     protected NextAction onFailureNoRetry(Packet packet, CallResponse<V1NamespaceList> callResponse) {
       return isNotAuthorizedOrForbidden(callResponse)
-          ? doNext(packet) : super.onFailureNoRetry(packet, callResponse);
+          ? doNext(createDomainCrdAndStartNamespaces(targetNamespaces), packet) : 
```

Comment:
Sorry, I did not notice this earlier. 
I am not sure what this change is for. It tries to create domain CRD and start the namespaces if isNotAuthorizedOrForbidden() returns true. Not sure this is correct. in addition, this change does not seem to relate to non-existing namespaces.

Reply:
Previously create domain CRD and start namespaces steps were called before the async call to read existing namespaces (unconditionally). I made this change to cover the case where  async call fails and comes to "onFailureNoRetry" after multiple retries. We still execute create domain CRD and start namespaces steps before execute next step. If failure is due to some reason other than 401 or 403, it calls "super.onFailureNoRetry". I'm not sure if calling create domain CRD and start namespaces steps will be useful here. The async call is just to read  existing namespaces and if it fails and goes onFailureNoRetry after multiple retries, there's some fundamental problem with cluster.

@rjeberhard and @doxiao  - Please let me know if this looks incorrect or if you have any suggestions. Thanks.

Reply:
Just had a discussion with Ryan about this change and it looks fine. Namespace is global k8s resource and some customers don't want operator to have permission on global resources. So we still want to go ahead and create CRD and start namespaces when API call to list namespaces fails due to authorization error. In other failure cases, operator pod liveness probe will fail and pod will be restarted by kubernetes. So create domain CRD and start namespace will be executed when pod restarts.

Reply:
Okay. It sounds fine. thanks.